src/components/media-views.js

```diff
@@ -761,7 +766,11 @@ AFRAME.registerComponent("media-image", {
 
       let cacheItem;
       if (textureCache.has(src)) {
-        cacheItem = textureCache.retain(src);
+        if (this.currentSrcIsRetained) {
```

Comment:
Hmm, a bit surprised if this is ever happening...

Reply:
i guess for images it would only happen if update gets called for the same src, which seems unlikely currently -- because it'd have to be if the content type changed. i think this code is more correct though, and covers the future potential cases where there is more state on this component that can be mutated. (eg perhaps an image filter)